views/default/output/color.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/**
+ * Displays the input color
+ *
+ * @uses $vars['value'] Color code in HexCode format
+ * @uses $vars['class']
+ * @uses $vars['background-color']
+ */
+
+$value = elgg_extract('value', $vars);
+
+if (!preg_match('/^#[a-f0-9]{6}$/i', $value)) {
+	return;
+}
+
+$vars['class'] = elgg_extract_class($vars, 'elgg-color-box');
+$vars['style'] = "background-color: $value";
+
+echo elgg_format_element('span', $vars) . " " . elgg_view("output/text", strtoupper($vars));
```

Comment:
the reason I am using strtoupper is because it will be show the hex code in a consistence format. So be it in small letter or capital letter the hex value will always show in upper case.